docs/sources/clients/promtail/troubleshooting.md

```diff
@@ -19,6 +19,31 @@ To start Promtail in dry run mode use the flag `--dry-run` as shown in the examp
 cat my.log | promtail --stdin --dry-run --client.url http://127.0.0.1:3100/loki/api/v1/push
 ```
 
+## Inspecting
+
+Promtail can be configured to print all changes to log entries as each pipeline stage is executed (via the flag `--inspect`).
+Each log entry contains four fields:
+- line
+- timestamp
+- labels
+- extracted fields
+
+Whenever any of these change, a log message will be printed:
+
+```bash
+cat my.log | promtail --stdin --dry-run --inspect --client.url http://127.0.0.1:3100/loki/api/v1/push
+```
+
+![inspect.png](inspect.png)
+
+It can be used in combination with `--stdin` and `--dry-run`.
+
+Additions will be highlighted in green, modifications in yellow, and removals in red.
+
+If no changes are applied during a stage, that is usually an indication of a misconfiguration or undesired behaviour.
+
+**Note**: the `--inspect` flag should _not_ be used in production.
```

Comment:
I don't think that this needs to be a "Note."  It doesn't hurt anything, but it does draw a reader's eyes away from all the other important prose before the Note.  If it should not be missed by readers, then I would suggest changing the very first sentence of the section to be:
> For environments other than production environments, Promtail can output all changes to log entries as each pipeline stage is executed.

Can you add a phrase or sentence about why the inspect should not be used in production?

Reply:
Good point, I will do :+1: